### PR TITLE
Add Redact Transition Snippet

### PR DESCRIPTION
**Pull request by benhayehudi**
This adds the snippet for redacting a transaction.

It parallels the existing PHP snippet on NDP.

### CODE REVIEW

File: redact/redact.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'dotenv'
+Dotenv.load
+
+NEXMO_API_KEY = ENV['NEXMO_API_KEY']
+NEXMO_API_SECRET = ENV['NEXMO_API_SECRET']
+NEXMO_REDACT_ID = ENV['REDACT_ID']
+
+NEXMO_REDACT_TYPE = ARGV[0]
```
Comment (mheap) [inline redact/redact.rb]: Why is this a parameter rather than an environment variable?

File: redact/redact.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'dotenv'
+Dotenv.load
+
+NEXMO_API_KEY = ENV['NEXMO_API_KEY']
+NEXMO_API_SECRET = ENV['NEXMO_API_SECRET']
+NEXMO_REDACT_ID = ENV['REDACT_ID']
```
Comment (mheap) [inline redact/redact.rb]: This should be called `NEXMO_REDACT_ID` in the environment

File: redact/redact.rb (hunk 1 above)
Comment (benhayehudi) [reply redact/redact.rb]: As with the verify snippets already in production, it made sense to me to make this a dynamically provided argument as there are multiple options a user could provide for redact type, why not let it be defined when executing? I liked that pattern in the verify snippets, and implemented it here as well.

File: redact/redact.rb (hunk 2 above)
Comment (benhayehudi) [reply redact/redact.rb]: 👍

File: redact/redact.rb (hunk 1 above)
Comment (mheap) [reply redact/redact.rb]: We can do that, but in that case I'd expect `NEXMO_REDACT_ID` to be a parameter too as they're both variable

Comment (benhayehudi) [reply redact/redact.rb]: Yes, good idea. I'll make the change now.

File: redact/redact.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+require 'dotenv'
+require 'byebug'
+Dotenv.load
+
+NEXMO_API_KEY = ENV['NEXMO_API_KEY']
+NEXMO_API_SECRET = ENV['NEXMO_API_SECRET']
+
+NEXMO_REDACT_ID = ARGV[0]
+if NEXMO_REDACT_ID.empty?
```
Comment (mheap) [inline redact/redact.rb]:
```
Traceback (most recent call last):
redact/redact.rb:11:in `<main>': undefined method `empty?' for nil:NilClass (NoMethodError)
```

**Comment by benhayehudi** [comment]
@mheap - bumping up again for a review 🙏

File: redact/redact.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  exit
+end
+
+NEXMO_REDACT_TYPE = ARGV[1] || ''
+if NEXMO_REDACT_TYPE.empty?
+  puts 'Please supply the product type'
+  exit
+end
+
+byebug
```
Comment (mheap) [inline redact/redact.rb]: ❗️

Comment (benhayehudi) [reply redact/redact.rb]: 😱

Comment (benhayehudi) [reply redact/redact.rb]: This has been removed!

**Pull request merged**